Cannot build and run for x64 MacOS (Catalina etc)

**Describe the bug**
Cannot compile and run for x64 (macos)

**To Reproduce**

```
 cmake -DOPENSSL_ROOT_DIR="/usr/local/opt/openssl" -G "Ninja" -A x64
 cmake -DOPENSSL_ROOT_DIR="/usr/local/opt/openssl" -G "Xcode" -A x64
```

Both yield:

```
  Generator

    Ninja/Xcode

  does not support platform specification, but platform

    x64

  was specified.
```

The readme says Ninja *only* builds for x64, though.

**Expected behavior**
Builds for x64.

**Screenshots**

I tried running ninja w/o the x64 flag and it build this:

![image](https://user-images.githubusercontent.com/19196536/76955611-3a0d1680-695e-11ea-80bf-e271e8e1b1f3.png)

![image](https://user-images.githubusercontent.com/19196536/76955623-3e393400-695e-11ea-88fc-72f1e165af25.png)

Not sure what this means.

**Desktop (please complete the following information):**
![image](https://user-images.githubusercontent.com/19196536/76955674-56a94e80-695e-11ea-9d9c-ad88161ccb23.png)

2017 macbook (not pro), latest catalina, x64 (they dropped x32 support).

**Additional context**

I tried to build it again. Here is the output:

```
cmake -DOPENSSL_ROOT_DIR="/usr/local/opt/openssl" -G "Ninja"
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error at CMakeLists.txt:14 (project):
  Generator

    Ninja

  does not support platform specification, but platform

    x64

  was specified.


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_ASM_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "/Users/lachlan/etterna/CMakeFiles/CMakeOutput.log".
```

Hey, thanks for replying.

1. Yes, that's right. I can generate the binary (I am not sure how I did this when I opened the original issue, I thought I used ninja but now I'm thinking it was via Xcode)
2. I have been running my `make` commands from the root (not in build). I have a strong suspicion I was mistaken in my original issue. I make a directory called `build`, `cd` in there and ran this (with `make`. Should it be `cmake`? I put the `cmake` output after this:

## with `make` in the `build` directory

```
(master) $ make -DOPENSSL_ROOT_DIR=$(which openssl) -G "Ninja"
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- D
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- O
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- P
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- E
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- G
Usage: make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
  --debug[=FLAGS]             Print various types of debugging information.
  -e, --environment-overrides
                              Environment variables override makefiles.
  -f FILE, --file=FILE, --makefile=FILE
                              Read FILE as a makefile.
  -h, --help                  Print this message and exit.
  -i, --ignore-errors         Ignore errors from commands.
  -I DIRECTORY, --include-dir=DIRECTORY
                              Search DIRECTORY for included makefiles.
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
  -k, --keep-going            Keep going when some targets can't be made.
  -l [N], --load-average[=N], --max-load[=N]
                              Don't start multiple jobs unless load is below N.
  -L, --check-symlink-times   Use the latest mtime between symlinks and target.
  -n, --just-print, --dry-run, --recon
                              Don't actually run any commands; just print them.
  -o FILE, --old-file=FILE, --assume-old=FILE
                              Consider FILE to be very old and don't remake it.
  -p, --print-data-base       Print make's internal database.
  -q, --question              Run no commands; exit status says if up to date.
  -r, --no-builtin-rules      Disable the built-in implicit rules.
  -R, --no-builtin-variables  Disable the built-in variable settings.
  -s, --silent, --quiet       Don't echo commands.
  -S, --no-keep-going, --stop
                              Turns off -k.
  -t, --touch                 Touch targets instead of remaking them.
  -v, --version               Print the version number of make and exit.
  -w, --print-directory       Print the current directory.
  --no-print-directory        Turn off -w, even if it was turned on implicitly.
  -W FILE, --what-if=FILE, --new-file=FILE, --assume-new=FILE
                              Consider FILE to be infinitely new.
  --warn-undefined-variables  Warn when an undefined variable is referenced.
  -N OPTION, --NeXT-option=OPTION
                              Turn on value of NeXT OPTION.

This program built for i386-apple-darwin11.3.0
Report bugs to <bug-make@gnu.org>
(master) $
```

## with `cmake` in `build`

```
(master) $ cmake -DOPENSSL_ROOT_DIR=$(which openssl) -G "Ninja"
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error: The source directory "/Users/lachlan/etterna/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
(master) $
```

3. About Xcode, I am not opening the project in the Xcode app, I ran it from the command line using cmake and the `-G Xcode` flag.

Maybe I'm doing something obviously wrong? I am not very familiar with make or building C projects. I also realized I'm doing this on master - maybe that was the problem, I can try again on develop (should master be stable for macos x64?)

--- a/src/arch/Threads/Threads_Pthreads.cpp
+++ b/src/arch/Threads/Threads_Pthreads.cpp
@@ -340,11 +340,10 @@ EventImpl_Pthreads::Wait(RageTimer* pTimeout)
 		/* If we support condattr_setclock, we'll set the condition to use
 		 * the same clock as RageTimer and can use it directly. If the
 		 * clock is CLOCK_REALTIME, that's the default anyway. */
-		uint64_t usec = pTimeout->c_dur.count();
-		unsigned sec = floor(usec / 1000000);
-		unsigned nsec = (usec - sec * 1000000) * 1000;
-		abstime.tv_sec = sec;
-		abstime.tv_nsec = nsec;
+		auto nsec = std::chrono::duration_cast<std::chrono::nanoseconds>(pTimeout->c_dur);
+		nsec -= std::chrono::duration_cast<std::chrono::seconds>(pTimeout->c_dur);
+		abstime.tv_sec = pTimeout->c_dur.count();
+		abstime.tv_nsec = nsec.count();
 	} else {
 		// The RageTimer clock is different than the wait clock; convert it.
 		timeval tv;
@@ -355,11 +354,10 @@ EventImpl_Pthreads::Wait(RageTimer* pTimeout)
 		float fSecondsInFuture = -pTimeout->Ago();
 		timeofday += fSecondsInFuture;
 
-		uint64_t usec = timeofday.c_dur.count();
-		unsigned sec = floor(usec / 1000000);
-		unsigned nsec = (usec - sec * 1000000) * 1000;
-		abstime.tv_sec = sec;
-		abstime.tv_nsec = nsec;
+		auto nsec = std::chrono::duration_cast<std::chrono::nanoseconds>(timeofday.c_dur);
+		nsec -= std::chrono::duration_cast<std::chrono::seconds>(timeofday.c_dur);
+		abstime.tv_sec = pTimeout->c_dur.count();
+		abstime.tv_nsec = nsec.count();
 	}
 
 	int iRet = pthread_cond_timedwait(&m_Cond, &m_pParent->mutex, &abstime);

